core/store/src/lib.rs

```diff
@@ -190,9 +190,47 @@ pub fn create_store(path: &str) -> Arc<Store> {
     Arc::new(Store::new(db))
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum StorageError {
+    /// Key-value db internal failure
+    StorageInternalError,
+    /// Storage is PartialStorage and requested a missing trie node
+    TrieNodeMissing,
+    /// Either invalid state or key-value db is corrupted.
+    /// For PartialStorage it cannot be corrupted.
+    /// Error message is unreliable and for debugging purposes only. It's also probably ok to
+    /// panic in every place that produces this error.
+    /// We can check if db is corrupted by verifying everything in the state trie.
+    StorageInconsistentState(String),
```

Comment:
This looks like it should be `StorageInconsistentState(StorageInconsistentState)` where `StorageInconsistentState` is an enum on its own that implements `Display` that's where we should have string errors like `"Failed to deserialize"`.

Reply:
The idea is that StorageInconsistentState means something went terribly wrong so its types don't need to be machine readable and there isn't a well defined classification

Reply:
Ok